tests/e2e/env/setup.sh

```diff
@@ -1,366 +1,520 @@
+		server_vars+=('WCP_SERVER_REPO')
+		server_vars+=('WCP_SERVER_BRANCH')
+		server_vars+=('E2E_WCPAY_STRIPE_TEST_PUBLIC_KEY')
+		server_vars+=('E2E_WCPAY_STRIPE_TEST_SECRET_KEY')
+	elif [[ "$E2E_USE_LOCAL_SERVER" = false ]]; then
+		server_vars+=('E2E_BLOG_TOKEN')
+		server_vars+=('E2E_USER_TOKEN')
 	fi
 
-	cd "$SERVER_PATH"
-
-	step "Creating server secrets"
-	SECRETS="<?php
-	define( 'WCPAY_STRIPE_TEST_PUBLIC_KEY', '$E2E_WCPAY_STRIPE_TEST_PUBLIC_KEY' );
-	define( 'WCPAY_STRIPE_TEST_SECRET_KEY', '$E2E_WCPAY_STRIPE_TEST_SECRET_KEY' );
-	define( 'WCPAY_STRIPE_TEST_WEBHOOK_SIGNATURE_KEY', '$E2E_WCPAY_STRIPE_TEST_WEBHOOK_SIGNATURE_KEY' );
-	define( 'WCPAY_STRIPE_LIVE_PUBLIC_KEY', 'pk_live_XXXXXXX' );
-	define( 'WCPAY_STRIPE_LIVE_SECRET_KEY', 'sk_live_XXXXXXX' );
-	define( 'WCPAY_ONBOARDING_ENCRYPT_KEY', str_repeat( 'a', SODIUM_CRYPTO_SECRETBOX_KEYBYTES ) );
-	"
-	printf "$SECRETS" > "local/secrets.php"
-	echo "Secrets created"
-
-	step "Starting SERVER containers"
-	redirect_output docker-compose -f docker-compose.yml -f docker-compose.e2e.yml up --build --force-recreate -d
-
-	# Get WordPress instance port number from running containers, and print a debug line to show if it works.
-	WP_LISTEN_PORT=$(docker ps | grep "$SERVER_CONTAINER" | sed -En "s/.*0:([0-9]+).*/\1/p")
-	echo "WordPress instance listening on port ${WP_LISTEN_PORT}"
-
-	if [[ -n $CI ]]; then
-		echo "Setting docker folder permissions"
-		redirect_output sudo chown www-data:www-data -R ./docker/wordpress
-		redirect_output ls -al ./docker
+	# Define required client variables.
+	local client_vars=(
+		'WCP_DEV_TOOLS_REPO'
+		'WCP_DEV_TOOLS_BRANCH'
+		'E2E_WP_VERSION'
+		'E2E_WC_VERSION'
+		'SKIP_WC_SUBSCRIPTIONS_TESTS'
+		'SKIP_WC_ACTION_SCHEDULER_TESTS'
+		'SKIP_WC_BLOCKS_TESTS'
+	)
+
+	# Define other required variables.
+	local other_vars=(
+		'E2E_GH_TOKEN'
+		'FORCE_E2E_DEPS_SETUP'
+	)
+
+	# Check if Docker is running.
+	if ! docker info >/dev/null 2>&1; then
+		add_log "dependency_logs" "$(colorize_yellow '- Docker service is not running.')"
+		((error_count++))
 	fi
 
-	step "Setting up SERVER containers"
-	"$SERVER_PATH"/local/bin/docker-setup.sh
-
-	step "Configuring server with stripe account"
-	"$SERVER_PATH"/local/bin/link-account.sh "$BLOG_ID" "$E2E_WCPAY_STRIPE_ACCOUNT_ID" test 1 1
+	# Check for script dependencies.
+	if ! command -v jq &> /dev/null; then
+		add_log "dependency_logs" "$(colorize_yellow '- jq library is missing.')"
+		((error_count++))
+	fi
 
-	if [[ -n $CI ]]; then
-		step "Disable Xdebug on server container"
-		docker exec "$SERVER_CONTAINER" \
-		sh -c 'echo "#zend_extension=xdebug" > /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini && echo "Xdebug disabled."'
+	# Loop through server variables and check if they are set.
+	check_vars "server_logs" "${server_vars[@]}"
+	check_vars "client_logs" "${client_vars[@]}"
+	check_vars "other_logs" "${other_vars[@]}"
+
+	# Log to terminal if any errors are present and exit.
```

Comment:
How does storing all log messages in an array help? Is it only to be able to group errors by category and to be able to have the red message in the beginning, instead of in the end? If yes, is the complexity worth it?

Reply:
So bash supports associative arrays, which can be used to store the logs, but that doesn't work with Bash versions < 4. 

I used the current approach as an alternative to keep the logs contained to a single place for each category and to keep the code clean. 

Let me know if you have other suggestions?

Reply:
And yes, I wanted to keep the error text order.

Reply:
Just printing as an errors comes up sounds as good and also makes the code so much simpler.

Reply:
Sure. I'll make modifications :) Thanks

Reply:
Thanks! And wanted to follow up on the changes, I don't see more commits.

Reply:
@nb I had to shift focus to WooPayments MC architecture project which is priority for our team at the moment.

So I'll addressing the reviews sometime next week.